apps/web/src/dialogs/command-palette/command-palette-dialog.tsx

```diff
@@ -0,0 +1,277 @@
+                            bg:
+                              command.index === selected
+                                ? "hover"
+                                : "transparent",
+                            ".chip": {
+                              bg:
+                                command.index === selected
+                                  ? "color-mix(in srgb, var(--accent) 20%, transparent)"
+                                  : "var(--background-secondary)"
+                            },
+                            ":hover:not(:disabled):not(:active)": {
+                              bg: "hover"
+                            }
+                          }}
+                        >
+                          {command.icon && (
+                            <command.icon
+                              size={18}
+                              color={
+                                command.index === selected
+                                  ? "icon-selected"
+                                  : "icon"
+                              }
+                            />
+                          )}
+                          {["note", "notebook", "reminder", "tag"].includes(
+                            command.type
+                          ) ? (
+                            <Text
+                              className="chip"
+                              sx={{
+                                px: 1,
+                                borderRadius: "4px",
+                                border: "1px solid",
+                                borderColor: "border"
+                              }}
+                            >
+                              <Highlighter text={command.title} query={query} />
+                            </Text>
+                          ) : (
+                            <Highlighter text={command.title} query={query} />
+                          )}
+                        </Button>
+                      ))}
+                    </Flex>
+                  </Flex>
+                ))}
+              </Flex>
+            </ScrollContainer>
+          </>
+        </Flex>
+        <Flex
+          sx={{ flexDirection: "row", bg: "hover", justifyContent: "center" }}
+        >
+          <Text variant="subBody" sx={{ m: 0.5 }}>
+            <kbd>{">"}</kbd> for command mode · remove <kbd>{">"}</kbd> for
+            search mode · <kbd>⏎</kbd> to select · <kbd>↑</kbd>
+            <kbd>↓</kbd> to navigate
+          </Text>
+        </Flex>
+      </Dialog>
+    );
+  }
+);
+
+function Highlighter({ text, query }: { text: string; query: string }) {
+  const queryClean = query.startsWith(">")
+    ? query.slice(1).trim()
+    : query.trim();
+  const result = queryClean.length > 0 ? match(queryClean, text) : false;
+
+  return (
+    <span
+      dangerouslySetInnerHTML={{
+        __html:
+          result && result.match
+            ? surround(text, {
+                result: result,
```

Comment:
I have to use fuzzyjs's `match` again here to be able to use `surround`. Is this fine, or do you think we should refactor so that the `lookup` in `core` returns the object that `surround` expects?

Reply:
I've added a `db.lookup.fuzzy` method. Not sure if that was exactly what you meant, do take a look please